execution-plan/src/memory.rs

```diff
@@ -270,6 +270,12 @@ impl Memory {
             .last()
     }
 
+    /// Get the address of the last empty cell.
+    /// If none, then all cells are occupied.
+    pub fn next_empty_cell(&self) -> Option<usize> {
+        self.iter().position(|(_, v)| v.is_none())
```

Comment:
might be a bit slow to iterate through all the addresses for large programs, but, whatever. Large programs probably have more API calls that'll take up time anyway.

Reply:
This is similar efficiency as `last_nonempty_address` above, no?

Reply:
In the future this can easily be optimized by tracking the end of the stack